### ISSUE TITLE
Packages need some love

### ISSUE THREAD

**Issue by davidfowl** [dotnet/corefxlab#451]
These packages aren't .NET Core friendly:

https://github.com/dotnet/corefxlab/tree/master/nuget
- They should be using the new netstandard (dotnet TFM) right now. 
- They should list their dependencies. The project.json lists them but the nuspec does not.
- They target portable+\* still, which is fine for now but won't be in the long run.
- aspnetcore50 isn't a thing anymore so it should be removed.

@ericstj can help with some of this

**Comment by KrzysztofCwalina** [comment]
@sokket

**Comment by sokket** [comment]
If we use dotnet CLI to generate the NuSpec files dynamically, will that fix some of these issues?

**Comment by davidfowl** [comment]
If you use the CLI to produce packages it'll do the right thing. 

/cc @livarcocc @krwq Can you guys help set this up? 

I'm going to make some tweaks to pack to produce symbol packages

**Comment by tarekgh** [comment]
@sokket are you willing to fix all nuspec files in corefxlab? or should every owner fix his own library?

**Comment by sokket** [comment]
There is some weirdness with the branch right now due to, I think, our release yesterday. I'm trying to get the branch green and then I can work with @livarcocc to get all nuspec files working properly.

**Comment by davidfowl** [comment]
Ditch the nuspec and just use the CLI. It unifies producing and consuming a package so you'll never have to worry about this problem again. Look at how we build *all* of ASP.NET https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging/project.json

**Comment by sokket** [comment]
@livarcocc are we in a state where we can do this today?

**Comment by livarcocc** [comment]
Right now, there are only three projects that fail to build with the CLI:

1. System.Text.Utf8, due to the custom compiler it needs to use. I haven't figured out yet how to make that work. I saw that @davidfowl has a change to pick the compiler first in the folder, and I have been thinking about using that but haven't had a chance to try yet.
2. System.Reflection.Metadata.Cil: Which I have been having problems to move it into project.json and make it still work with msbuild. Almost there.
3. System.Drawing.Graphics, which has a problem with conditional files to be included in the project.json depending on the OS. Haven't looked into this yet.

All the other ones should be working with the CLI, at least building.

**Comment by sokket** [comment]
Awesome! I think we need to wait for System.Text.Utf8 to be building first since other packages rely on that. Metadata.Cil and Graphics aren't pushed to myget today so I won't be blocked on those. @livarcocc could please let me know when you figure out the Utf8 issue? Then I can do a change to make the nuspecs rely on dotnet cli instead of hardcoded files

@davidfowl When using nuget pack with project.json, I assume we cans till override the version that is in the project.json, correct? Right now we use the $version$ placeholder in the XML, it'd be great if that would still work in the project.json world :)

**Comment by KrzysztofCwalina** [comment]
@livarcocc, one of the things dotnet-cli project was promising was that the tools are be pluggable. Is plugging in a custom front end compiler not supported by dotnet-cli?

**Comment by sokket** [comment]
@davidfowl I updated our project.json files but NuGet doesn't seem to pick them up when doing:

`nuget.exe pack <project.json>`

**Comment by davidfowl** [comment]
@sokket NuGet doesn't support `project.json` packing yet. The dotnet cli does however. That code will be moving into nuget in the coming weeks. Right now `dotnet pack` does a compile followed by a package (in process). It will eventually change to be compile + nuget pack once it supports project.json.

You should also move all of the metadata from the nuspec to project.json and change dnxcore50 to dotnet5.4

**Comment by sokket** [comment]
I think once Livar finishes his work, this should come for free...assigning over to him

**Comment by livarcocc** [comment]
Jon, assigning this back to you. I saw that you are doing this in your latest PR.

**Comment by sokket** [comment]
Cool, this should be resolved when PR #581 is merged

**Comment by borgdylan** [comment]
Also, can the packages use the 1.6.0 version of NETStandard.Library instead of the rc2 bits?

**Comment by joshfree** [comment]
closing out old infra / packaging bug that's no longer relevant.
